url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/frontend/handlers.java

```diff
@@ -0,0 +1,959 @@
+                        try
+                        {
+                            out.close();
+                        } catch (IOException ex)
+                        {
+                             FrontEnd.appendToPane(FrontEnd.statuswindow,"Error Occured...\n",Color.BLACK);
+                            FrontEnd.appendToPane(FrontEnd.statuswindow,ex.getStackTrace().toString(),Color.BLACK);
+//                             FrontEnd.statuswindow.append("Error occured ...\n");
+//                            FrontEnd.statuswindow.append(ex.getStackTrace().toString());
+                            //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                            //frontend.FrontEnd.exceptionraised++;
+                        }
+                    }
+
+                    temp.deleteOnExit();
+
+                } catch (IOException e)
+                {
+                }
+            }
+
+        } catch (Exception ex)
+        {
+            Logger.getLogger(FrontEnd.class.getName()).log(Level.SEVERE, null, ex);
+        }
+
+      }
+
+    public static void update_memorytable(int memref_ct, int ch)
+      {
+        Object[][] obj = backend.Memory.display(ch);
+
+        clean_memtable();
+        for (int i = 0; i < memref_ct; i++)
+        {
+            memorytable.model2.addRow(obj[i]);
+            //System.err.println("updating mem table");
+
+        }
+      }
+
+    public static void clean_branchtable()
+      {
+        int c = JumpTable.model2.getRowCount();
+        for (int i = c - 1; i >= 0; i--)
+        {
+            JumpTable.model2.removeRow(i);
+
+        }
+      }
+
+    public static void clean_memtable()
+      {
+        int c = memorytable.model2.getRowCount();
+        for (int i = c - 1; i >= 0; i--)
+        {
+            memorytable.model2.removeRow(i);
+
+        }
+      }
+
+    public void open(File file)
+      {
+        int ob;
+        if ((ob = check_if_open(file.getAbsolutePath())) != -1)
+        {
+            FrontEnd.EditorPane.setSelectedIndex(ob);
+        } else
+        {
+            backend.Register.resetRegisters();
+            try
+            {
+                FrontEnd.activepane = new JEditorPane();
+                JScrollPane scroll = new JScrollPane();
+//                   scroll.setHorizontalScrollBarPolicy(ScrollPaneConstants.HORIZONTAL_SCROLLBAR_ALWAYS);
+                scroll.setViewportView(FrontEnd.activepane);
+                FrontEnd.activepane.setEditorKit(new NumberedEditorKit());
+                FrontEnd.activepane.setHighlighter(new frontend.MyHighlighter());
+               // FrontEnd.activepane.setSelectionColor(new Color(1.0f, 1.0f, 1.0f, 0.0f));//line replaced by nikhil in place of below on saturday 3:45;
+               FrontEnd.activepane.setSelectionColor(new Color(0.0f, 0.0f, 1.0f, 0.1f));
+
+                FrontEnd.activepane.read(new FileReader(file.getAbsoluteFile()), null);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileReader*. Currently, there are execution paths that do not contain closure statements, for example, when JTextComponent.read throws an exception. Ensure the object returned by *FileReader()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).